biz.aQute.launchpad/test/aQute/launchpad/BundleBuilderTest.java

```diff
@@ -0,0 +1,88 @@
+package aQute.launchpad;
+
+import org.assertj.core.api.JUnitSoftAssertions;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.osgi.framework.Bundle;
+
+import aQute.bnd.osgi.Verifier;
+
+public class BundleBuilderTest {
+	LaunchpadBuilder	builder;
+	Launchpad			lp;
+	BundleBuilder 		bb;
+	
+	@Rule
+	public JUnitSoftAssertions	softly	= new JUnitSoftAssertions();
+
+
+	@Before
+	public void before() throws Exception {
+		builder = new LaunchpadBuilder();
+		try {
+			lp = builder.runfw("org.apache.felix.framework")
+				.create();
+		} catch (Exception e) {
+			builder.close();
+		}
+		bb = lp.bundle();
+	}
+
+	@After
+	public void after() throws Exception {
+		try {
+			bb.close();
```

Comment:
You can also use `IO.close(bb)` as `IO.close` will not throw any exception.

Reply:
Such a simple suggestion which led me down a small rabbit warren...
* First problem: `BundleBuilder` does not implement `AutoCloseable`. In fact, its `close()` method was package private, and wasn't even called anywhere, ever. I've now made it public and made it implement `AutoCloseable` in this PR.
* Second problem: `IO.close()` only takes a `Closeable`, whereas `Launchpad` and `LaunchpadBuilder` (and now `BundleBuilder`) all implement `AutoCloseable` rather than `Closeable`. I tried to change the method signature, but Bnd complained that I had changed the API. Instead, I added IO.close(AutoCloseable) and changed IO.close(Closeable) to simply call the former. Had to bump the version of the package - wasn't sure whether to bump minor or micro; I bumped micro but that is easily fixed if it would be better to bump minor instead - please let me know (not that experienced with versioning, yet).

If necessary/appropriate, I can split both of these changes out into one or two separate PR.